src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1515 @@
+                if self.config.unpad_no_grad:
+                    input_ids, indices, cu_seqlens, max_seqlen, *_ = self._unpad_inputs_no_grad(
+                        input_ids, attention_mask
+                    )
+                else:
+                    input_ids, indices, cu_seqlens, max_seqlen, *_ = self._unpad_inputs(input_ids, attention_mask)
+        elif position_ids is None:
+            position_ids = torch.arange(seq_len, device=input_ids.device).unsqueeze(0)
+
+        hidden_states = self.embeddings(input_ids)
+
+        # expand attention_mask
+        if self.config._attn_implementation != "flash_attention_2" and attention_mask is not None:
+            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+            attention_mask = _prepare_4d_attention_mask(attention_mask, hidden_states.dtype)
+
+        for encoder_layer in self.layers:
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            if self.gradient_checkpointing and self.training:
+                layer_outputs = self._gradient_checkpointing_func(
+                    encoder_layer.__call__,
+                    hidden_states,
+                    attention_mask,
+                    position_ids,
+                    cu_seqlens,
+                    max_seqlen,
+                    output_attentions,
+                )
+            else:
+                layer_outputs = encoder_layer(
+                    hidden_states,
+                    attention_mask=attention_mask,
+                    position_ids=position_ids,
+                    cu_seqlens=cu_seqlens,
+                    max_seqlen=max_seqlen,
+                    output_attentions=output_attentions,
+                )
+            hidden_states = layer_outputs[0]
+            if output_attentions and len(layer_outputs) > 1:
+                all_self_attentions = all_self_attentions + (layer_outputs[1],)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        hidden_states = self.final_norm(hidden_states)
+
+        if repad:
+            hidden_states, _ = self._pad_outputs(hidden_states, indices, batch_size, seq_len)
+
+        if not return_dict:
+            return tuple(v for v in [hidden_states, all_hidden_states, all_self_attentions] if v is not None)
+        return BaseModelOutput(
+            last_hidden_state=hidden_states,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+
+class ModernBertForMaskedLM(ModernBertPreTrainedModel):
+    _tied_weights_keys = ["decoder.weight"]
+
+    def __init__(self, config: ModernBertConfig):
+        super().__init__(config)
+        self.config = config
+        self.model = ModernBertModel(config)
+        self.head = ModernBertPredictionHead(config)
+
+        if config.tie_word_embeddings:
+            decoder_weights = self.model.embeddings.tok_embeddings.weight
```

Comment:
the way to do this is by defining the `    _tied_weights_keys = ["model.embeddings.tok_embeddings.weight"]`

Reply:
Fixed in b444c15e3936036a7a1100b44fedb61f0c923261